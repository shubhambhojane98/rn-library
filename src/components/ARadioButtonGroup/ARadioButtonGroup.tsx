import {View, TouchableOpacity} from 'react-native';
import React, {FC, useState} from 'react';
import ARadioButton from '../ARadioButton/ARadioButton';

interface RadioData {
  id: number;
  value: string;
}

interface Props {
  data: RadioData[];
  onSelectValue: (radio: RadioData) => void;
}

const ARadioButtonGroup: FC<Props> = ({data, onSelectValue}) => {
  const [selectedItem, setSelectedItem] = useState({id: -1});

  const selectHandler = (item: any) => {
    setSelectedItem(item);
    onSelectValue(item);
  };

  return (
    <View>
      {data.map((item, index) => {
        return (
          <TouchableOpacity
            key={index}
            onPress={() => selectHandler(item)}
            activeOpacity={1}>
            <ARadioButton
              label={item.value}
              selected={selectedItem.id === item.id}
            />
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default ARadioButtonGroup;
