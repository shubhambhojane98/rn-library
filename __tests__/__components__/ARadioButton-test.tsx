import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ARadioButtonGroup from '../../src/components/ARadioButtonGroup/ARadioButtonGroup';

const RadioData = [
  {id: 1, value: 'First'},
  {id: 2, value: 'Second'},
  {id: 3, value: 'Third'},
  {id: 4, value: 'Four'},
  {id: 5, value: 'Five'},
];

it('renders correctly with default prop', () => {
  const tree = renderer
    .create(<ARadioButtonGroup onSelectValue={jest.fn} data={RadioData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
