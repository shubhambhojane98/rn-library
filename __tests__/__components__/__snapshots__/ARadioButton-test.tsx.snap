// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default prop 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        height={20}
        name="unselectedradio"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#231F20",
              "fontSize": 19.65714285714286,
              "marginHorizontal": 6.142857142857142,
            },
          ]
        }
      >
        First
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        height={20}
        name="unselectedradio"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#231F20",
              "fontSize": 19.65714285714286,
              "marginHorizontal": 6.142857142857142,
            },
          ]
        }
      >
        Second
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        height={20}
        name="unselectedradio"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#231F20",
              "fontSize": 19.65714285714286,
              "marginHorizontal": 6.142857142857142,
            },
          ]
        }
      >
        Third
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        height={20}
        name="unselectedradio"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#231F20",
              "fontSize": 19.65714285714286,
              "marginHorizontal": 6.142857142857142,
            },
          ]
        }
      >
        Four
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <SvgMock
        height={20}
        name="unselectedradio"
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 21.13142857142857,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#231F20",
              "fontSize": 19.65714285714286,
              "marginHorizontal": 6.142857142857142,
            },
          ]
        }
      >
        Five
      </Text>
    </View>
  </View>
</View>
`;
